File: map-api/src/main/kotlin/domain/model/LineVersion.kt
package cz.cvut.fit.gaierda1.domain.model

@JvmInline
value class LineId(val value: String)

data class LineVersion(
    val lineId: LineId,
    val publicCode: String,
    val name: String,
    val shortName: String,
    val transportMode: String, // TODO make enum
    val validIn: DateRange,
    val isDetour: Boolean,
) {
}

File: map/src/api/model/lineVersion.ts
type LineVersion = {
    lineId: string,
    publicCode: string,
    name: string,
    shortName: string,
    transportMode: string,
    validFrom: string,
    validTo: string,
    isDetour: boolean,
}

type LineVersionWithDates = {
    lineId: string,
    publicCode: string,
    name: string,
    shortName: string,
    transportMode: string,
    validFrom: Date,
    validTo: Date,
    isDetour: boolean,
}

export type { LineVersion, LineVersionWithDates }
